File: src/main/java/io/mcdocker/launcher/content/clients/ClientManifest.java
/*
 * MCDocker, an open source Minecraft launcher.
 * Copyright (C) 2021 MCDocker
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package io.mcdocker.launcher.content.clients;

public class ClientManifest {

    private String name;
    private int javaVersion;
    private String mainClass;
    private final String type;

    public ClientManifest(String name, int javaVersion, String mainClass, String type) {
        this.name = name;
        this.javaVersion = javaVersion;
        this.mainClass = mainClass;
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getJavaVersion() {
        return javaVersion;
    }

    public void setJavaVersion(int javaVersion) {
        this.javaVersion = javaVersion;
    }

    public String getType() {
        return type;
    }

    public String getMainClass() {
        return mainClass;
    }

    public void setMainClass(String mainClass) {
        this.mainClass = mainClass;
    }

}

File: bin/content/client/client_manifest.dart
class ClientManifest {
  String name;
  int javaVersion;
  String mainClass;
  String startupArguments;
  final String type;

  ClientManifest(this.name, this.javaVersion, this.mainClass,
      this.startupArguments, this.type);

  ClientManifest.fromJson(Map<String, dynamic> json)
      : name = json['name'],
        javaVersion = json['javaVersion'],
        mainClass = json['mainClass'],
        startupArguments = json['startupArguments'],
        type = json['type'];

  Map<String, dynamic> toJson() => {
        'name': name,
        'javaVersion': javaVersion,
        'mainClass': mainClass,
        'startupArguments': startupArguments,
        'type': type
      };
}
